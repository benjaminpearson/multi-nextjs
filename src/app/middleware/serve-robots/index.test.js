import serveRobots from '.';

describe('server', () => {
  it('serves robots when /robots.txt requested', () => {
    const req = { url: '/robots.txt' };
    const res = { setHeader: jest.fn(), send: jest.fn() };
    const result = serveRobots({ req, res });
    expect(res.setHeader).toHaveBeenCalledWith('content-type', 'text/plain');
    expect(res.send.mock.calls).toMatchSnapshot();
    expect(result).toBe(true);
  });

  it.todo('serves robots with robots disallowed when non production');

  it.todo('serves robots with robots allowed when production');

  it('does not serve robots when req url does not equal /robots.txt', () => {
    expect(serveRobots({ req: { url: '/foo' } })).toBe(false);
  });
})

describe('client', () => {
  it('does not serve robots when req is empty', () => {
    expect(serveRobots({})).toBe(false);
  });
})
