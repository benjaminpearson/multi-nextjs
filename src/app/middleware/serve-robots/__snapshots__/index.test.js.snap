// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server serves robots when /robots.txt requested 1`] = `
Array [
  Array [
    "User-agent: *
Disallow: /
Sitemap: http://www.example.com/sitemap.xml",
  ],
]
`;
